python/langsmith/utils.py

```diff
@@ -252,6 +252,26 @@ def is_base_message_like(obj: object) -> bool:
     )
 
 
+def get_tracer_project(return_default_value=True) -> Optional[str]:
+    """Get the project name for a LangSmith tracer."""
+    return os.environ.get(
+        # Hosted LangServe projects get precedence over all other defaults.
+        # This is to make sure that we always use the associated project
+        # for a hosted langserve deployment even if the customer sets some
+        # other project name in their environment.
+        "HOSTED_LANGSERVE_PROJECT_NAME",
+        os.environ.get(
+            "LANGCHAIN_PROJECT",
+            os.environ.get(
+                # This is the legacy name for a LANGCHAIN_PROJECT, so it
+                # has lower precedence than LANGCHAIN_PROJECT
+                "LANGCHAIN_SESSION",
```

Comment:
We should probably start actually warning users for this, it's a really old (1 yr) variables so i don't think it's used in practies

Reply:
Do you want that as part of this PR, or something different?